### ISSUE TITLE
Error with Angular Ivy compiler

### ISSUE THREAD

**Issue by ignlg** [Cordobo/angularx-qrcode#63]
Angular version: `8.1.0`

Ivy compiler enabled in `tsconfig.json` with:

```
  "angularCompilerOptions": {
    "enableIvy": true
  },
```

Error reported:

```
ERROR in Attempted to get members of a non-class: "class QRCodeComponent {
    constructor(el, platformId) {
        this.el = el;
        this.platformId = platformId;
        /** @internal */
        this.allowEmptyString = false;
        this.colordark = '#000000';
        this.colorlight = '#ffffff';
        this.level = 'M';
        this.hidetitle = false;
        this.qrdata = '';
        this.size = 256;
        this.usesvg = false;
        this.isValidQrCodeText = (data) => {
            if (this.allowEmptyString === false) {
                return !(typeof data === 'undefined' || data === '');
            }
            return !(typeof data === 'undefined');
        };
    }
    ngAfterViewInit() {
        if (isPlatformServer(this.platformId)) {
            return;
        }
        if (!QRCode) {
            QRCode = require('qrcodejs2');
        }
        try {
            if (!this.isValidQrCodeText(this.qrdata)) {
                throw new Error('Empty QR Code data');
            }
            this.qrcode = new QRCode(this.el.nativeElement, {
                colorDark: this.colordark,
                colorLight: this.colorlight,
                correctLevel: QRCode.CorrectLevel[this.level.toString()],
                height: this.size,
                text: this.qrdata || ' ',
                useSVG: this.usesvg,
                width: this.size,
            });
        }
        catch (e) {
            console.error('Error generating QR Code: ' + e.message);
        }
    }
    ngOnChanges(changes) {
        if (!this.qrcode) {
            return;
        }
        const qrData = changes['qrdata'];
        if (qrData && this.isValidQrCodeText(qrData.currentValue)) {
            this.qrcode.clear();
            this.qrcode.makeCode(qrData.currentValue);
        }
    }
}"
```

**Comment by softsimon** [comment]
same here

> ERROR in getInternalNameOfClass() called on a non-ES5 class: expected QRCodeComponent to have an inner class declaration

**Comment by AndreasGassmann** [comment]
Same issue here with version `1.7.0-beta.5`. Did anyone find a workaround?

**Comment by Cordobo** [comment]
Related question, <Maybe<Solution>>

https://github.com/angular/angular/issues/34182

**Comment by Slash7GNR** [comment]
Hey,
Can anyone update ? Any idea when the library will be ready for ivy ?

Thanks.

**Comment by kklindic** [comment]
Works fine with Ivy, implemented on `Angular 9.0.7`.

**Comment by Cordobo** [comment]
Hi,

I fixed this issue today in the latest release (2.2.1). A working sample app is available for angular 9.1:

https://github.com/Cordobo/angularx-qrcode-sample-app

For angular 9.x run

npm install angularx-qrcode@^2.2.1
or
yarn addl angularx-qrcode@^2.2.1
